ngramoffset.go

```diff
@@ -356,3 +357,89 @@ func (a *asciiNgramOffset) DumpMap() map[ngram]simpleSection {
 func (a *asciiNgramOffset) SizeBytes() int {
 	return 4*len(a.entries) + 4*len(a.chunkOffsets)
 }
+
+// ngramMap is an transient type while we investigate the performance of
+// combinedNgramOffset (established) vs binarySearch (new).
+//
+// It is like an interface, but we do the drudgery so we still get a useful
+// zero value (instead of nil panics in tests).
+type ngramMap struct {
+	offsetMap combinedNgramOffset
+	bsMap     binarySearchNgram
+}
+
+func (m ngramMap) Get(gram ngram) simpleSection {
+	if m.offsetMap.asc != nil {
+		return m.offsetMap.Get(gram)
+	}
+	return m.bsMap.Get(gram)
+}
+
+func (m ngramMap) DumpMap() map[ngram]simpleSection {
+	if m.offsetMap.asc != nil {
+		return m.offsetMap.DumpMap()
+	}
+	return m.bsMap.DumpMap()
+}
+
+func (m ngramMap) SizeBytes() int {
+	if m.offsetMap.asc != nil {
+		return m.offsetMap.SizeBytes()
+	}
+	return 0 // binarySearch only uses mmaped data.
+}
+
+type binarySearchNgram struct {
+	// ngramText is the bytes at indexTOC.ngramText
+	//
+	// It is a sorted ngramSlice marshalled as list of bigendian uint64s.
+	ngramText []byte
+	// postingIndex is the index section of the compoundSection for the posting
+	// lists.
+	//
+	// It is a list of offsets in the a order corresponding with ngramText. It
+	// is marshalled as a list of bigendian uint32s.
+	postingOffsets []uint32
+	// postingDataSentinalOffset is where postingData ends in the index file.
+	// This is used to calculate the size of the last posting.
+	postingDataSentinalOffset uint32
```

Comment:
s/Sentinal/Sentinel/

Reply:
If I understood correctly, this file uses the ```toc.postings.relativeIndex()``` trick to add such sentinels but for experiment it doesn't matter.